### PR TITLE
Add `FIRST` & `LAST` as aliases to `MAX_BY` & `MIN_BY`

### PR DESCRIPTION
**Pull request by matriv**
## Summary of the changes / Why this improves CrateDB

Follows: https://github.com/crate/crate/pull/13127

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -53,8 +53,9 @@ None
 Changes
 =======
 
-- Added the :ref:`MAX_BY <aggregation-max_by>` and :ref:`MIN_BY
-  <aggregation-min_by>` aggregation functions
+- Added the :ref:`MAX_BY <aggregation-max_by>` (alias
+  :ref:`FIRST <aggregation-first>`) and :ref:`MIN_BY <aggregation-min_by>`
+  (alias :ref:`LAST <aggregation-last>`) aggregation functions.
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:58]: Isn't LAST == MAX_BY and FIRST == MIN_BY?

**Comment by mfussenegger** [comment]
What's the motivation for this?

File: docs/appendices/release-notes/unreleased.rst (hunk 1 above)
Comment (matriv) [reply docs/appendices/release-notes/unreleased.rst]: very true! thx!

**Comment by matriv** [comment]
> What's the motivation for this?

FIRST/LAST are more popular in timeseries DBs, which is the main use case to introduce those aggs.
I've posted on https://github.com/crate/crate/issues/10528 and @proddata and @smu agreed it's nice to add.

**Comment by mfussenegger** [comment]
Strong no from my side.

We don't aim for compatibility with any of the systems using `FIRST` / `LAST`.

I went with `max_by` and `min_by` because:
 - It's closer to the existing `max` and `min` aggregation than the existing `first_value` / `last_value` window functions.
 - It's also used by others (Trino and DuckDB - although it is an alias in the latter)
 
 If `first` / `last` is better by any metric we should rename them, not add an alias. Having multiple names just makes the API surface larger and the product more complex. There should be good reasons to do something like that (e.g. compatibility with PostgreSQL, we don't target other systems)

**Comment by matriv** [comment]
Thx for the explanation @mfussenegger!
@seut @proddata ok to close this then?

**Comment by matriv** [comment]
Won't proceed with these aliases, see: https://github.com/crate/crate/pull/13134#issuecomment-1277483212

**Comment by proddata** [comment]
I think `max_by`/ `min_by` is fine, as this is a more powerful version, of what others are offering (i.e. it is not only working with time(stamp) columns)

**Pull request closed without merging**